painters/submission_jxs.py

```diff
@@ -0,0 +1,117 @@
+from os.path import join
+
+import numpy as np
+from data_provider import SUBMISSION_INFO_FILE, DATA_DIR
+from data_provider import load_organized_data_info
+from utils import append_to_file
+from utils import read_lines_in_batches
+from keras.models import model_from_json
+from os.path import basename
+
+IMGS_DIM_1D = 256
+SUBMISSION_FILE = join(DATA_DIR, 'submission.csv')
+BATCH_SIZE = 10000
+FILES_TO_AVG = {}
+
+
+def _create_submission_file_avg_cnns():
+    data_info = load_organized_data_info(IMGS_DIM_1D)
+    model = _get_model()
+    X_test, names = _average_embedded_test_data(model, data_info)
+    features_lookup = {n: f for n, f in zip(names, X_test)}
+    _create_submission_file(
+        BATCH_SIZE, features_lookup, _calculate_batch_prediction_dot)
+
+
+def _get_model():
+    with open('../models/model.json', 'r') as json_file:
+        model = model_from_json(json_file.read())
+    model.load_weights('../models/weights.h5')
+    return model
+
+
+def _average_embedded_test_data(model, data_info):
+    X_test, y_test = None, None
+
+    data_info = load_organized_data_info(IMGS_DIM_1D)
+    dir_te, num_te = data_info['dir_te'], data_info['num_te']
+    dir_tr = data_info['dir_tr']
+    from data_provider import testing_generator, init_directory_generator
+    gen = testing_generator(dir_tr=dir_tr)
+    gen_test = init_directory_generator(
+        gen, dir_te, BATCH_SIZE, class_mode='sparse', shuffle_=False)
+
+    num_full_epochs = num_te // BATCH_SIZE
+    last_batch_size = num_te - (num_full_epochs * BATCH_SIZE)
```

Comment:
`num_full_epochs` is a little confusing of a variable name here -- does it refer to "number of batches in an epoch"?